messaging/src/main/java/org/axonframework/commandhandling/gateway/DefaultReactiveCommandGateway.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandCallback;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.GenericCommandMessage;
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.Registration;
+import org.axonframework.messaging.ReactiveMessageDispatchInterceptor;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static java.util.Arrays.asList;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Default implementation of the {@link ReactiveCommandGateway}.
+ *
+ * @author Milan Savic
+ * @since 4.4
+ */
+public class DefaultReactiveCommandGateway implements ReactiveCommandGateway {
```

Comment:
Is this the `DefaultReactiveCommandGateway` or a `ReactorReactiveCommandGateway`? The implementation "assumption" this makes, is that you use Project Reactor, right?

Reply:
That's also something that's worth mentioning in the class level Javadoc